medical/models/medical_abstract_entity.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import threading
+
+from openerp import api, fields, models
+
+
+class MedicalAbstractEntity(models.AbstractModel):
+    _name = 'medical.abstract.entity'
+    _description = 'Medical Abstract Entity'
+    _inherits = {'res.partner': 'partner_id'}
+
+    # Redefine ``active`` so that it is managed independently from partner.
+    active = fields.Boolean(
+        default=True,
+    )
+    partner_id = fields.Many2one(
+        string='Related Partner',
+        comodel_name='res.partner',
+        required=True,
+        ondelete='cascade',
+        index=True,
+    )
+    type = fields.Selection(
+        default=lambda s: s._name,
+        related='partner_id.type',
+    )
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, vals):
+        vals = self._create_vals(vals)
+        return super(MedicalAbstractEntity, self).create(vals)
+
+    @api.multi
+    def action_invalidate(self):
+        """ It deactivates a patient, and their partner if no active entities.
+        """
+        for record in self:
+            record.active = False
+            entities = self.env[self._name].search([
+                ('partner_id', 'child_of', record.partner_id.id),
+                ('parent_id', 'child_of', record.partner_id.id),
+                ('active', '=', True),
+            ])
+            if not entities:
+                record.partner_id.active = False
+
+    @api.multi
+    def _compute_identification(self, field_name, category_code, many=False):
+        """ It computes a field that indicates a certain ID type.
+
+        Use this on a field that represents a certain ID type. It will compute
+        the desired field as that ID(s).
+
+        Set many to ``True`` to allow many results. This can only be used on
+        m2m fields.
+
+        Example:
+
+            .. code-block:: python
+
+            social_security_id = fields.Many2one(
+                string='Social Security',
+                comodel_name='res.partner.id_number',
+                compute=lambda s: s._compute_identification(
+                    'social_security_id', 'SSN',
+                )
+            )
+            npi_ids = fields.Many2many(
+                string='NPI Numbers',
+                comodel_name='res.partner.id_number',
+                compute=lambda s: s._compute_identification(
+                    'npi_ids', 'NPI', True,
+                )
+            )
+
+        Args:
+            field_name: Name of field to set.
+            category_code: Category code of the Identification type.
+            many: Allow many results.
+        """
+        for record in self:
+            ids = record.id_numbers.filtered(
+                lambda r: r.category_id.code == category_code
+            )
+            if not ids:
+                continue
+            value = [(6, 0, ids)] if many else ids[0]
```

Comment:
Oh wait I remember the reasoning behind this now - it will be possible for this to be manipulated out from underneath the record & thus break the record. 

This is because the `id_numbers` is actually in the partner, but we are performing this pass-thru in the entity. It would also be possible by just creating a `res.partner.id_number` record outside of the partner if we weren't on the entity though, just less common. 

I proposed this feature to OCA/partner-contact, so we might be able to get a bit better handling if it gets moved there. 

Consider the following as an example:

* User adds a second driver's license through partner, but leaves the other active
* User navigates to partner's associated patient record
* UserError(*)